packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -0,0 +1,1114 @@
+								);
+
+								task.title = "Pushed all slices";
+							},
+						},
+						{
+							title: "Pushing custom types...",
+							skip: () => {
+								if (!this.options.push) {
+									return `--no-push used`;
+								} else if (!this.options.pushCustomTypes) {
+									return `--no-push-custom-types used`;
+								}
+							},
+							task: async (_, task) => {
+								const { ids, errors } =
+									await this.manager.customTypes.readCustomTypeLibrary();
+
+								if (errors.length > 0) {
+									// TODO: Provide better error message.
+									throw new Error(
+										`Failed to read custom type libraries: ${errors.join(
+											", ",
+										)}`,
+									);
+								}
+
+								if (!ids || ids.length === 0) {
+									task.skip("No custom type to push");
+
+									return;
+								}
+
+								let pushed = 0;
+								task.title = `Pushing custom types... (0/${ids.length})`;
+								await Promise.all(
+									ids.map(async (id) => {
+										await this.manager.customTypes.pushCustomType({ id });
+										pushed++;
+										task.title = `Pushing custom types... (${pushed}/${ids.length})`;
+									}),
+								);
+
+								task.title = "Pushed all custom types";
+							},
+						},
+						{
+							title: "Pushing documents...",
+							skip: () => {
+								if (!this.options.push) {
+									return `--no-push used`;
+								} else if (!this.options.pushDocuments) {
+									return `--no-push-documents used`;
+								}
+							},
+							task: async (_, task) => {
+								assertExists(
+									this.context.repository,
+									"Repository selection must be available through context to run `pushDataToPrismic`",
+								);
+
+								const root = await this.manager.project.getRoot();
+								const documentsDirectoryPath = path.resolve(root, "documents");
+
+								try {
+									await fs.access(documentsDirectoryPath);
+								} catch {
+									parentTask.title = "Pushed data to Prismic";
+									task.skip("No document to push");
+
+									return;
+								}
+
+								const signaturePath = path.resolve(
+									documentsDirectoryPath,
+									"index.json",
+								);
+								const rawSignature = await fs.readFile(signaturePath, "utf-8");
+								const signature: string = JSON.parse(rawSignature).signature;
+
+								const documentsGlob = await globby("*/*.json", {
+									cwd: documentsDirectoryPath,
+								});
+								if (documentsGlob.length === 0) {
+									parentTask.title = "Pushed data to Prismic";
+									task.skip("No document to push");
+
+									return;
+								}
+
+								const documents: [string, unknown][] = await Promise.all(
+									documentsGlob.map(async (document) => {
+										const filename = path.basename(document, ".json");
+										const fileContent = await fs.readFile(
+											path.resolve(documentsDirectoryPath, document),
+											"utf-8",
+										);
+
+										return [filename, JSON.parse(fileContent)];
```

Comment:
Maybe `catch` failed JSON parsing here?

Reply:
Good idea. I added a check in this commit: https://github.com/prismicio/slice-machine/commit/f0e2433fda792f665d537a8e514e30179809a3b5#diff-9bd9b7346b4e79315a4216e41c958e36d13508345728e46c470ec083271d3be5R949-R960